### ISSUE TITLE
Bookmark import/export via html

### ISSUE THREAD

**Issue by kufue** [fork-maintainers/iceraven-browser#103]
Title says it all

### What is the user problem or growth opportunity you want to see solved?

I never saw a happy firefox.apk user because bookmarks are useless, in focus bookmarks can be exported through newline clipboard text, but it seems as if humans didn't develop this browser at all, because only important features are, clickable buttons, and exporting research to computer, while not uploading to an American server (sync), this may be possible through an addon

**Comment by CharmCityCrab** [comment]
I agree that importing and exporting bookmarks in a standard file format would be a good feature to have.  I was really surprised when I went to Firefox for Android to export my bookmarks for use in Iceweasel and found out that I couldn't export them from Firefox and, if I had been able to, I wouldn't be able to import them into Iceweasel anyway because they inherited the issue from Firefox when they forked.

I don't know how feasible it would be to implement this suggestion, but I see it as a pro-user and sort of in line with the idea that people increasingly use mobile browsers the way they use desktop browsers because of larger screens, more powerful phone hardware, and so on and so forth.  I feel like some of the big players are using mobile browsers as a way to lock people into an ecosystem with their brand names all over it, typically via the cloud, and ideally (for the corporations that own the browsers) tying into some sort of pay service.  As an independent browser with volunteer maintainers, Iceweasel has no reason to focus on cloud servers it doesn't have, trying to create a lock-in to a desktop product it doesn't have (Although it can sync to Firefox), or trying to push to pay services it also doesn't have.

Being able to import and export bookmarks in standard formats is a routine feature of desktop browsers, and the reasons I suspect it isn't always available on mobile browsers don't apply to Iceweasel and are not in the user's best interest.  A sync option is nice for those who want it, but for many people, controlling their own bookmark folders and files is better, so that should be there too if possible.

This is unlikely to be something Mozilla implements soon on Firefox, so if Iceweasel wants the feature, someone will have to code it themselves (I am not saying it has to be a non-maintainer by any means, I'm saying we won't get it from upstream, is all), most likely.  I don't know how doable that is.  One thought I had is that if it is too complex to code from scratch, maybe a shim could be created that piggybacks on sync by essentially creating a "sync to yourself option", where the bookmark data that would otherwise be assembled for sync from the existing code is exported as a bookmarks file.

I also wonder if an open-source web extension exists that handles bookmarks and could be built into browser the code somehow to give Iceweasel it's own native bookmark system (i.e. Not the same as Firefox) that has an export/import feature.

I may be getting a little grandiose for the scope of a fork like this, though.  I'll leave it to the developers as to whether any of this is stuff they feel like they would be able to create and maintain.

**Comment by interfect** [comment]
This would be cool to have.

**Comment by MonicaQvQ** [comment]
Hello, can you realize this function?

**Comment by AnonBlankzy** [comment]
Yep, import/export bookmarks is the most important option for me. Whatever Firefox fork does this first,  I'll be using it.

**Comment by akliuxingyuan** [comment]
Firefox for Android Nightly version can import bookmarks from a file now.
